test/cpp/jit/test_lite_interpreter.cpp

```diff
@@ -1371,5 +1372,41 @@ static auto reg =
 
 } // namespace
 
+TEST(LiteInterpreterTest, OperatorCacheDifferentiatesDefaultArgs) {
+  // Create 3 methods:
+  //
+  // 1. forward() returns a tensor with dtype=torch.int64 (4)
+  // 2. forward2() returns a tensor with dtype=torch.float32 (6)
+  // 3. forward3() returns a tensor with dtype=torch.float32 but
+  //    the dtype is inferred by the input tensor's dtype
```

Comment:
"inferred by the input tensor's dtype"
you mean the input scalar passed to fill_ (i.e. the 12.0) --I guess ti gets transformed to a tensor rank 0?

Reply:
@raziel `new_empty` deduces the output tensors dtype based on the input tensor's dtype (not the value passed in to `fill_`). I think `fill_` will just cast the input into the output dtype of the tensor it is trying to fill.

```
>>> torch.zeros(10).new_empty([10])
tensor([-3.8947e-10,  4.5874e-41, -3.8947e-10,  4.5874e-41, -4.7881e-10,
         4.5874e-41, -4.7838e-10,  4.5874e-41, -4.7838e-10,  4.5874e-41])
>>> torch.zeros(10, dtype=torch.int64).new_empty([10])
tensor([140607294414080, 140607294414080,              32,              64,
         94720912317360,               0,               0,               0,
                      0,               0])
>>> torch.zeros(10).new_empty([10]).fill_(52)
tensor([52., 52., 52., 52., 52., 52., 52., 52., 52., 52.])
>>> torch.zeros(10, dtype=torch.int64).new_empty([10]).fill_(54.2)
tensor([54, 54, 54, 54, 54, 54, 54, 54, 54, 54])
```

Reply:
In this case, the input tensor is the one returned by `torch.zeros(10)`, which happens to have dtype=float32 (that seems to be the default if none is specified).